TestTxSimulator/blob_sequence flake

```
8:24PM INF starting gRPC server... address=127.0.0.1:44202
{"level":"warn","errorMessage":"tls: first record does not look like a TLS handshake","time":"2026-02-09T20:24:38Z","message":"failed to connect with the config to grpc server; proceeding with insecure connection"}
--- FAIL: TestTxSimulator (153.23s)
    --- FAIL: TestTxSimulator/blob_sequence (1.03s)
        run_test.go:126: 
            	Error Trace:	/home/runner/work/celestia-app/celestia-app/test/txsim/run_test.go:126
            	Error:      	Should be true
            	Test:       	TestTxSimulator/blob_sequence
            	Messages:   	sequence 0: querying account tx-sim-3: rpc error: code = NotFound desc = account celestia1xdmdf5ypdwczfxhfjtru65dywc5dyx7e245ln3 not found
        network.go:97: tearing down testnode
FAIL
FAIL	github.com/celestiaorg/celestia-app/v7/test/txsim	163.160s
FAIL
make: *** [Makefile:280: test] Error 1
Error: Process completed with exit code 2.
```

https://github.com/celestiaorg/celestia-app/actions/runs/21839223127/job/63018680650

--- a/test/txsim/account.go
+++ b/test/txsim/account.go
@@ -20,9 +20,12 @@ import (
 	"github.com/cosmos/cosmos-sdk/crypto/hd"
 	"github.com/cosmos/cosmos-sdk/crypto/keyring"
 	"github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
 	"github.com/rs/zerolog/log"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 type AccountManager struct {
@@ -371,6 +374,12 @@ func (am *AccountManager) GenerateAccounts(ctx context.Context) error {
 		return fmt.Errorf("error funding accounts: %w", err)
 	}
 
+	// Wait until all funded accounts are queryable via the auth gRPC endpoint
+	// to avoid a race between tx confirmation and account state availability.
+	if err := am.waitForAccounts(ctx); err != nil {
+		return fmt.Errorf("error waiting for accounts to be queryable: %w", err)
+	}
+
 	// print the new accounts
 	for _, acc := range am.pending {
 		am.accountIndex++
@@ -383,6 +392,37 @@ func (am *AccountManager) GenerateAccounts(ctx context.Context) error {
 
 	// clear the pending accounts
 	am.pending = nil
+	return nil
+}
+
+// waitForAccounts polls the auth gRPC endpoint until each pending account is
+// queryable. This avoids a race where ConfirmTx reports the funding tx as
+// committed but the ABCI Query endpoint hasn't yet indexed the new account.
+func (am *AccountManager) waitForAccounts(ctx context.Context) error {
+	authClient := authtypes.NewQueryClient(am.conn)
+	for _, acc := range am.pending {
+		addr := acc.address.String()
+		ticker := time.NewTicker(am.pollTime)
+		for {
+			select {
+			case <-ctx.Done():
+				ticker.Stop()
+				return ctx.Err()
+			case <-ticker.C:
+				_, err := authClient.Account(ctx, &authtypes.QueryAccountRequest{Address: addr})
+				if err == nil {
+					ticker.Stop()
+					goto next
+				}
+				if s, ok := status.FromError(err); ok && s.Code() == codes.NotFound {
+					continue
+				}
+				ticker.Stop()
+				return fmt.Errorf("error querying account %s: %w", addr, err)
+			}
+		}
+	next:
+	}
 	return nil
 }
 

